src/main/java/org/folio/event/service/UserEventProducer.java

```diff
@@ -40,39 +41,47 @@ public UserEventProducer(KafkaConfig kafkaConfig) {
+    UserEvent event = getUserEvent(user, tenantId, isPersonalDataChanged, eventAction);
     String eventPayload = Json.encode(event);
 
-    switch (eventAction) {
-      case CREATE:
-        logger.info("Starting to send user created event with id: {} for User to Kafka for userId: {}", event.getId(), user.getId());
-        return sendToKafka(UserEventType.USER_CREATED, eventPayload, okapiHeaders, user.getId());
-      case DELETE:
-        logger.info("Starting to send user deleted event with id: {} for User to Kafka for userId: {}", event.getId(), user.getId());
-        return sendToKafka(UserEventType.USER_DELETED, eventPayload, okapiHeaders, user.getId());
-      case EDIT:
-        logger.info("Starting to send user edit event with id: {} for User to Kafka for userId: {}", event.getId(), user.getId());
-        return sendToKafka(UserEventType.USER_UPDATED, eventPayload, okapiHeaders, user.getId());
-      default:
-        throw new IllegalStateException("Unexpected value: " + eventAction);
-    }
+      return switch (eventAction) {
+          case CREATE -> {
+              logger.info("Starting to send user created event with id: {} for User to Kafka for userId: {}", event.getId(), user.getId());
+              yield sendToKafka(UserEventType.USER_CREATED, eventPayload, okapiHeaders, user.getId());
+          }
+          case DELETE -> {
+              logger.info("Starting to send user deleted event with id: {} for User to Kafka for userId: {}", event.getId(), user.getId());
+              yield sendToKafka(UserEventType.USER_DELETED, eventPayload, okapiHeaders, user.getId());
+          }
+          case EDIT -> {
+              logger.info("Starting to send user edit event with id: {} for User to Kafka for userId: {}", event.getId(), user.getId());
+              yield sendToKafka(UserEventType.USER_UPDATED, eventPayload, okapiHeaders, user.getId());
+          }
+          default -> throw new IllegalStateException("Unexpected value: " + eventAction);
+      };
   }
 
-  private UserEvent getUserEvent(User user, String tenantId, UserEvent.Action eventAction) {
+  private UserEvent getUserEvent(User user, String tenantId, boolean isPersonalDataChanged, UserEvent.Action eventAction) {
     Metadata metadata = user.getMetadata();
     UserEvent event = new UserEvent();
     event.setId(UUID.randomUUID().toString());
     event.setAction(eventAction);
     event.setEventDate(new Date());
     event.setTenantId(tenantId);
+    event.setPersonalDataChanged(isPersonalDataChanged);
     if (Objects.nonNull(metadata)) {
       event.setActionDate(metadata.getCreatedDate());
       event.setPerformedBy(metadata.getUpdatedByUserId());
     }
     event.setUser(UsersService.getConsortiumUserDto(user));
+    if (ObjectUtils.notEqual(UserEvent.Action.DELETE, eventAction)) {
+      event.setBarcode(user.getBarcode());
+      event.setExternalSystemId(user.getExternalSystemId());
```

Comment:
Barcode and externalSystemId already exists in User payload. So we don't need to introduce duplicates in the kafka event payload